{
    "CVE_data_meta": {
        "ASSIGNER": "security-advisories@github.com",
        "ID": "CVE-2021-21330",
        "STATE": "PUBLIC",
        "TITLE": "Open redirect vulnerability in aiohttp"
    },
    "affects": {
        "vendor": {
            "vendor_data": [
                {
                    "product": {
                        "product_data": [
                            {
                                "product_name": "aiohttp",
                                "version": {
                                    "version_data": [
                                        {
                                            "version_value": "< 3.7.4"
                                        }
                                    ]
                                }
                            }
                        ]
                    },
                    "vendor_name": "aio-libs"
                }
            ]
        }
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
        "description_data": [
            {
                "lang": "eng",
                "value": "aiohttp is an asynchronous HTTP client/server framework for asyncio and Python. In aiohttp before version 3.7.4 there is an open redirect vulnerability. A maliciously crafted link to an aiohttp-based web-server could redirect the browser to a different website. It is caused by a bug in the `aiohttp.web_middlewares.normalize_path_middleware` middleware. This security problem has been fixed in 3.7.4. Upgrade your dependency using pip as follows \"pip install aiohttp >= 3.7.4\". If upgrading is not an option for you, a workaround can be to avoid using `aiohttp.web_middlewares.normalize_path_middleware` in your applications."
            }
        ]
    },
    "impact": {
        "cvss": {
            "attackComplexity": "HIGH",
            "attackVector": "NETWORK",
            "availabilityImpact": "NONE",
            "baseScore": 3.1,
            "baseSeverity": "LOW",
            "confidentialityImpact": "NONE",
            "integrityImpact": "LOW",
            "privilegesRequired": "NONE",
            "scope": "UNCHANGED",
            "userInteraction": "REQUIRED",
            "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:U/C:N/I:L/A:N",
            "version": "3.1"
        }
    },
    "problemtype": {
        "problemtype_data": [
            {
                "description": [
                    {
                        "lang": "eng",
                        "value": "CWE-601: URL Redirection to Untrusted Site ('Open Redirect')"
                    }
                ]
            }
        ]
    },
    "references": {
        "reference_data": [
            {
                "name": "https://github.com/aio-libs/aiohttp/security/advisories/GHSA-v6wp-4m6f-gcjg",
                "refsource": "CONFIRM",
                "url": "https://github.com/aio-libs/aiohttp/security/advisories/GHSA-v6wp-4m6f-gcjg"
            },
            {
                "name": "https://github.com/aio-libs/aiohttp/commit/2545222a3853e31ace15d87ae0e2effb7da0c96b",
                "refsource": "MISC",
                "url": "https://github.com/aio-libs/aiohttp/commit/2545222a3853e31ace15d87ae0e2effb7da0c96b"
            },
            {
                "name": "https://github.com/aio-libs/aiohttp/blob/master/CHANGES.rst#374-2021-02-25",
                "refsource": "MISC",
                "url": "https://github.com/aio-libs/aiohttp/blob/master/CHANGES.rst#374-2021-02-25"
            },
            {
                "name": "https://pypi.org/project/aiohttp/",
                "refsource": "MISC",
                "url": "https://pypi.org/project/aiohttp/"
            },
            {
                "refsource": "DEBIAN",
                "name": "DSA-4864",
                "url": "https://www.debian.org/security/2021/dsa-4864"
            },
            {
                "refsource": "FEDORA",
                "name": "FEDORA-2021-673b10ed77",
                "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/JN3V7CZJRT4QFCVXB6LDPCJH7NAOFCA5/"
            },
            {
                "refsource": "FEDORA",
                "name": "FEDORA-2021-902c1b07c9",
                "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FU7ENI54JNEK3PHEFGCE46DGMFNTVU6L/"
            },
            {
                "refsource": "GENTOO",
                "name": "GLSA-202208-19",
                "url": "https://security.gentoo.org/glsa/202208-19"
            }
        ]
    },
    "source": {
        "advisory": "GHSA-v6wp-4m6f-gcjg",
        "discovery": "UNKNOWN"
    }
}